sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosEndToEndOperationLatencyPolicyConfig.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.CosmosEndToEndOperationLatencyPolicyConfigBuilder;
+
+import java.time.Duration;
+
+/**
+ * Represents End to end operation latency policy config
+ * This enables requests to get cancelled by the client once the specified timeout is reached
+ */
+public class CosmosEndToEndOperationLatencyPolicyConfig {
```

Comment:
Name of this class has to be finalized. And all corresponding setters/getters have to be named accordingly

Reply:
Is `CosmosEndToEndOperationLatencyConfig` or we need `Policy` in it as well?

Reply:
I think we should have a config builder. You can't really make this class mutable (even add any mutable properties) - because otherwise you have to copy DEFAULT on every retrieval etc. Given that the config class has to be immutable it will result in ctor overload explosion if/when we add additional config. Take a look at the throughput control API - there Annie also added a builder for the configs for the same reason.

Reply:
There is a config builder class - `CosmosEndToEndOperationLatencyPolicyConfigBuilder`. But agree most of the methods in this class can change into private/protected